File: Rnwood.Smtp4dev/ApiClient/Message.ts
 
//MessageEntitySummary[] from Message
import MessageEntitySummary from './MessageEntitySummary';
//Header[] from Message
import Header from './Header';
export default class Message {

    constructor(id: string, from: string, to: string, receivedDate: Date, subject: string, parts: MessageEntitySummary[], headers: Header[], mimeParseError: string, ) {
         
        this.id = id; 
        this.from = from; 
        this.to = to; 
        this.receivedDate = receivedDate; 
        this.subject = subject; 
        this.parts = parts; 
        this.headers = headers; 
        this.mimeParseError = mimeParseError;
    }

     
    id: string; 
    from: string; 
    to: string; 
    receivedDate: Date; 
    subject: string; 
    parts: MessageEntitySummary[]; 
    headers: Header[]; 
    mimeParseError: string;
}

File: Rnwood.Smtp4dev.Shared/ApiModel/Message.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace Rnwood.Smtp4dev.ApiModel
{
    public class Message : ICacheByKey
    {
      

        public Guid Id { get; set; }

        public string From { get; set; }
        public string To { get; set; }
        public string Cc { get; set; }
        public string Bcc { get; set; }
        public DateTime ReceivedDate { get; set; }
        
        public bool SecureConnection { get; set; }

        public string Subject { get; set; }

        public List<MessageEntitySummary> Parts { get; set; } 

        public List<Header> Headers { get; set; }

        public string MimeParseError { get; set; }

        public string RelayError { get; set; }

        string ICacheByKey.CacheKey => Id.ToString();
    }
}
